2.4/root/usr/share/cont-docs/10-mongodb.txt

```diff
@@ -0,0 +1,124 @@
+`docker exec -ti CONTAINERID container-entrypoint` in a separate terminal.
+
+You can stop the detached container by running `docker stop CONTAINERID`.
+
+
+
+Database initialization
+-----------------------
+
+If the database directory is not initialized, the container script will create database users. If you mount already initialized database, script will skip this task. However you still have to specify '_USER+_PASSWORD+_DATABASE+_ADMIN_PASSWORD' environment variables to be able to connect to database from linked container.
+
+To pass arguments that are used for database initializing define them as environment variables (see table below).
+
+```
+docker run -d -e MONGODB_USER=user -e MONGODB_PASSWORD=pass -e MONGODB_DATABASE=db -e MONGODB_ADMIN_PASSWORD=adminpass -v /host/data:/var/lib/mongodb/data -p 27017:27017 THIS_IMAGE
+```
+
+This will create a container running MongoDB 2.6 with database
+`db` and user with credentials `user:pass` that has access to the database `db`.
+
+You can connect to the database by `docker run -ti THIS_IMAGE "mongo db -u user -p pass --host YOURIPADDRESS:27017"`.
+
+Users created by container script have these user roles:
+- admin user - `roles: ['dbAdminAnyDatabase', 'userAdminAnyDatabase' , 'readWriteAnyDatabase','clusterAdmin' ]`
+- MONGODB_USER - `roles: [ 'readWrite' ]` in database: `$MONGODB_DATABASE`
+
+Environment variables and volumes
+---------------------------------
+
+The image recognizes following environment variables that you can set during
+initialization, by passing `-e VAR=VALUE` to the Docker run command.
+
+|    Variable name          |    Description                              |
+| :------------------------ | ------------------------------------------- |
+|  `MONGODB_USER`           | User name for MongoDB account to be created |
+|  `MONGODB_PASSWORD`       | Password for the user account               |
+|  `MONGODB_DATABASE`       | Database name in which to create user       |
+|  `MONGODB_ADMIN_PASSWORD` | Password for the admin user                 |
+
+Following environment variables influence MongoDB configuration file. They are all optional.
+
+|    Variable name      |    Description                                                            |  Default  |
+| :-------------------- | ------------------------------------------------------------------------- | --------- |
+|  `MONGODB_NOPREALLOC` | Disable data file preallocation.                                          |  true     |
+|  `MONGODB_SMALLFILES` | Set MongoDB to use a smaller default data file size.                      |  true     |
+|  `MONGODB_QUIET`      | Runs MongoDB in a quiet mode that attempts to limit the amount of output. |  true     |
+
+You can also set following mount points by passing `-v /host:/container` flag to docker.
+
+|  Volume mount point      | Description            |
+| :----------------------- | ---------------------- |
+|  `/var/lib/mongodb/data` | MongoDB data directory |
```

Comment:
`/etc/mongodb.conf` is not specified as a volume in Dockerfile. So it is different to `/var/lib/mongodb/data`.
The mongodb.conf is mentioned in "configuration" section.